provision/kubernetes/deploy.go

```diff
@@ -433,17 +436,88 @@ func validateHC(hc *provTypes.TsuruYamlHealthcheck, client *ClusterClient) error
 		if hc.TimeoutSeconds >= hc.IntervalSeconds {
 			hc.IntervalSeconds = hc.TimeoutSeconds + 1
 		}
+
+		if err := ensureBackendConfig(ctx, client, app, hc); err != nil {
+			return err
+		}
+
 	}
 
 	return nil
 }
 
-func probesFromHC(hc *provTypes.TsuruYamlHealthcheck, client *ClusterClient, port int) (hcResult, error) {
+func int64PointerFromInt(v int) *int64 {
+	p := new(int64)
+	*p = int64(v)
+	return p
+}
+
+func ensureBackendConfig(ctx context.Context, client *ClusterClient, a provision.App, hc *provTypes.TsuruYamlHealthcheck) error {
+	exists, err := backendConfigCRDExists(ctx, client)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		return nil
+	}
+
+	backendConfigName := fmt.Sprintf("%s-backend-config", a.GetName())
```

Comment:
We should create a helper function for naming backend config similar to those in https://github.com/tsuru/tsuru/blob/6dd56808976bd8ee00dd35eac5a26f4dea288405/provision/kubernetes/helpers.go#L107-L110